client/blocks/upe/confirm-card-payment.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Handles the confirmation of card payments (3DSv2 modals/SCA challenge).
+ *
+ * @param {WCStripeAPI} api               The API used for connection both with the server and Stripe.
+ * @param {Object}   paymentDetails    Details about the payment, received from the server.
+ * @param {Object}   emitResponse      Various helpers for usage with observer response objects.
+ * @param {boolean}  shouldSavePayment Indicates whether the payment method should be saved or not.
+ * @return {Object}                An object, which contains the result from the action.
+ */
+export default async function confirmCardPayment(
+	api,
+	paymentDetails,
+	emitResponse,
+	shouldSavePayment
+) {
+	const { redirect, payment_method: paymentMethod } = paymentDetails;
+
+	try {
+		const confirmation = api.confirmIntent(
+			redirect,
+			shouldSavePayment ? paymentMethod : null
+		);
```

Comment:
This works even though `paymentMethod` is always `undefined` here, probably since we only need the payment method ID if we're actually saving the payment method? So maybe we can just replace the `shouldSavePayment ? paymentMethod : null` with `null` or `undefined`?

I don't fully understand what the API expects here, so I'm at a bit of a loss here; this is all copied from WCPay.

I'm honestly not really sure what this parameter is for